cmd/determinize-ci-operator/main.go

```diff
@@ -2,21 +2,36 @@ package main
 
 import (
 	"flag"
-	"os"
+	"fmt"
 
 	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/flagutil"
+	utilpointer "k8s.io/utils/pointer"
 
 	"github.com/openshift/ci-tools/pkg/api"
 	"github.com/openshift/ci-tools/pkg/config"
 )
 
-func gatherOptions() config.ConfirmableOptions {
-	o := config.ConfirmableOptions{}
-	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
-	o.Bind(fs)
-	if err := fs.Parse(os.Args[1:]); err != nil {
-		logrus.WithError(err).Fatal("could not parse input")
-	}
+type options struct {
+	config.ConfirmableOptions
+	migrateOpenshiftInstallerSRCTemplates   bool
+	templateMigrationCeiling                int
+	templateMigrationAllowedBranches        flagutil.Strings
+	templateMigrationAllowedOrgs            flagutil.Strings
+	templateMigrationAllowedClusterProfiles flagutil.Strings
+}
+
+func gatherOptions() options {
+	o := options{}
+	o.Bind(flag.CommandLine)
+	flag.BoolVar(&o.migrateOpenshiftInstallerSRCTemplates, "migrate-openshift-installer-src-templates", false, "If tests based on the openshift_installer_src template should be migrated to multi-step tests. What tests exactly to migrate can be limited further via the template-migration-allowed-{branch,org,cluster-profile} flags")
```

Comment:
Can we future-proof this a *tiny bit* and make it `--migrate-templates=openshift-installer-src,...` so this is easy to extend when we do the remaining ones?

Reply:
fixed